update readme (#1)
KseniyaKuzmina/how-to-calculate-routes-from-major-roads-using-the-bing-map-route-web-service-e5064: Readme.md

[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E5064)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

> # Map for WinForms - Use the Bing Map Route Web Service to Calculate Routes from Major Roads 

Reviewer: ```suggestion
# Map for WinForms - Use the Bing Maps Route Web Service to Calculate Routes from Major Roads
```

> This example demonstrates how to receive route information from the [Bing route service](https://docs.microsoft.com/en-us/bingmaps/rest-services/routes/) to calculate routes to the destination point from major roads, and how to display information about the resulting routes.

Reviewer: ```suggestion
This example demonstrates how to receive route information from the [Bing Maps route service](https://docs.microsoft.com/en-us/bingmaps/rest-services/routes/) to calculate routes to the destination point from major roads, and how to display information about the resulting routes.
```

![Resulting map](Images/resulting-map.png)

> > **NOTE:** You require to specify the Bing key to use Bing Maps services. Refer to the following tutorial for more information: [How to: Get a Bing Maps Key](https://docs.devexpress.com/WindowsForms/15102/controls-and-libraries/map-control/examples/general/how-to-get-a-bing-maps-key).

Reviewer: ```suggestion
> **NOTE:** You are required to specify the Bing Maps key to use Bing Maps services. Refer to the following tutorial for more information: [How to: Get a Bing Maps Key](https://docs.devexpress.com/WindowsForms/15102/controls-and-libraries/map-control/examples/general/how-to-get-a-bing-maps-key).
```

The main steps required to create such an application are described below.

Create a [`BingRouteDataProvider`](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.BingRouteDataProvider?p=netframework) object and assign it to the [InformationLayer.DataProvider](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.InformationLayer.DataProvider) property.

> Specify destination point coordinates ([GeoPoint.Latitude](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.GeoPoint.Latitude) and [GeoPoint.Longitude](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.GeoPoint.Longitude)). In addition, you can specify optional parameters: the destination name, driving or walking route travel mode, and route optimization options to calculate an optimal route either by time or by distance. To do this, set the [BingRouteOptions.Mode](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.BingRouteOptions.Mode) and [BingRouteOptions.RouteOptimization](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.BingRouteOptions.RouteOptimization) properties.

Reviewer: ```suggestion
Specify destination point coordinates ([GeoPoint.Latitude](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.GeoPoint.Latitude) and [GeoPoint.Longitude](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.GeoPoint.Longitude)). In addition, you can specify optional parameters (the destination name, driving or walking route travel mode, and route optimization options) to calculate the optimal route by time or by distance. To do this, set the [BingRouteOptions.Mode](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.BingRouteOptions.Mode) and [BingRouteOptions.RouteOptimization](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.BingRouteOptions.RouteOptimization) properties.
```

The provider generates [map items](https://docs.devexpress.com/WindowsForms/15091/controls-and-libraries/map-control/vector-data/vector-items?p=netframework) to display the route and waypoints since its [GenerateLayerItems](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.InformationDataProviderBase.GenerateLayerItems?p=netframework) property is enabled. You can handle the [InformationDataProviderBase.LayerItemsGenerating](https://docs.devexpress.com/WindowsForms/DevExpress.XtraMap.InformationDataProviderBase.LayerItemsGenerating?p=netframework) event to customize the item appearance.

## Files to Look At

* [Form1.cs](./CS/CalculateRoutesFromMajorRoads/Form1.cs) (VB: [Form1.vb](./VB/CalculateRoutesFromMajorRoads/Form1.vb))

## Documentation

> * [How to: Calculate Routes from Major Roads Using the Bing Route Service](https://docs.devexpress.com/WindowsForms/16666/controls-and-libraries/map-control/examples/gis-data/routing/how-to-calculate-routes-from-major-roads-using-the-bing-route-service?p=netframework)

Reviewer: Is it possible to rename this DevExpress topic title as follows?

`How to: Calculate Routes from Major Roads Using the Bing Maps Route Service`

* [Routing](https://docs.devexpress.com/WindowsForms/16713/controls-and-libraries/map-control/gis-data/routing?p=netframework)

## More Examples

* [How to zoom a map to a calculated route using the RouteCalculated event](https://github.com/DevExpress-Examples/how-to-zoom-a-map-to-a-calculated-route-using-the-routecalculated-event-t363647)
* [How to Implement a Custom Route Provider](https://github.com/DevExpress-Examples/how-to-implement-a-custom-route-provider-t410669)
* [How to implement a route calculator using GIS providers](https://github.com/DevExpress-Examples/how-to-implement-a-route-calculator-using-gis-providers-t187741)

> * [How to calculate a route between waypoints and customize the appearance of a route path using a Microsoft Bing Route web service](https://github.com/DevExpress-Examples/how-to-calculate-a-route-between-waypoints-and-customize-the-appearance-of-a-route-path-usin-e5100)

Reviewer: Can we update this GitHub Readme.md title as well?

`How to calculate a route between waypoints and customize the appearance of a route path using a Microsoft Bing Maps Route web service`
